build.py

```diff
@@ -120,28 +124,28 @@ def create_zip_file(source_dir, target_file):
         source_files = [os.path.basename(source_path)]
 
     # Copy them into the temporary directory.
-    cd(source_dir)
-    for file_name in source_files:
-        target_path = os.path.join(temp_dir, file_name)
-        target_dir = os.path.dirname(target_path)
-        if not os.path.exists(target_dir):
-            print('mkdir -p {}'.format(target_dir))
-            os.makedirs(target_dir)
-        print('cp {} {}'.format(file_name, target_path))
-        shutil.copyfile(file_name, target_path)
+    with cd(source_dir):
+        for file_name in source_files:
+            target_path = os.path.join(temp_dir, file_name)
+            target_dir = os.path.dirname(target_path)
+            if not os.path.exists(target_dir):
+                print('mkdir -p {}'.format(target_dir))
+                os.makedirs(target_dir)
+            print('cp {} {}'.format(file_name, target_path))
+            shutil.copyfile(file_name, target_path)
 
     # Install dependencies into the temporary directory.
     if runtime.startswith('python'):
         requirements = os.path.join(temp_dir, 'requirements.txt')
         if os.path.exists(requirements):
-            cd(temp_dir)
-            if runtime.startswith('python3'):
-                pip_command = 'pip3'
-            else:
-                pip_command = 'pip2'
-            run(pip_command, 'install', '-r', 'requirements.txt', '-t', '.')
+            with cd(temp_dir):
+                if runtime.startswith('python3'):
+                    pip_command = 'pip3'
+                else:
+                    pip_command = 'pip2'
+                run(pip_command, 'install', '-r', 'requirements.txt', '-t', '.')
 
     # Zip up the temporary directory and write it to the target filename.
     # This will be used by the Lambda function as the source code package.
-    create_zip_file(temp_dir, absolute_filename)
+    create_zip_file(temp_dir, os.path.splitext(absolute_filename)[0])
```

Comment:
I think the splitext -> basename logic is a weird requirement of shutil.make_archive so it should be done inside the create_zip_file function rather than here.

Reply:
`ext` will be `.zip` not `zip` in that example. The file name will always end in `.zip` because it's hardcoded in `hash.py` so checking the extension is probably overly cautious :)

Reply:
I like my helper functions to be unaware of such implementation details, if I can. :D